### PR TITLE
Adding Documentation

### PR DESCRIPTION
**Pull request by debog**
Added documentation to most parts of the code. I haven't yet gone through the demo case; I focused on the census case.

### CODE REVIEW

File: src/CaseData.cpp [hunk 1]
```diff
@@ -12,12 +16,41 @@
...
 {
     InitFromFile(fname);
 }
 
-void CaseData::InitFromFile (const std::string& fname)
+/*! \brief Read case data from a given file
+
+    The case data file is an ASCII text file with three columns of numbers:
+    FIPS code, current number of cases, and cumulative number of cases till date.
+    (QDG: But data/CaseData/July4.cases has 2 columns!)
```
Comment (debog) [inline src/CaseData.cpp]: @atmyers @tannguyen153 This function wants to read three columns from the file (FIPS code, current number of cases, and cumulative cases till date). But the example file (data/CaseData/July4.cases) has two columns.

File: src/DemographicData.cpp [hunk 2]
```diff
@@ -12,12 +16,37 @@
...
 {
     InitFromFile(fname);
 }
 
-void DemographicData::InitFromFile (const std::string& fname)
+/*! \brief Read in demographic data from given file.
+ *
+ *  + The first line of the file contains the number of units.
+ *  + The following lines have the following data:
+ *    + ID (QDG: what is this ID?)
```
Comment (debog) [inline src/DemographicData.cpp:30]: @atmyers @tannguyen153 What is the first column of the demographic data file? Also, is my understanding of the columns (as described here) correct?

File: src/DiseaseParm.H [hunk 3]
```diff
@@ -1,56 +1,79 @@
...
+    amrex::Real pSC = amrex::Real(0.2);     /*!< Contact matrix scaling factor - school */
+    amrex::Real pHSC = amrex::Real(0.3);    /*!< Contact matrix scaling factor - high school */
+    amrex::Real pMSC = amrex::Real(0.3);    /*!< Contact matrix scaling factor - middle school */
+    amrex::Real pESC = amrex::Real(0.3);    /*!< Contact matrix scaling factor - elementary school */
+    amrex::Real pCO = amrex::Real(1.45);    /*!< Contact matrix scaling factor - community */
+    amrex::Real pNH = amrex::Real(1.45);    /*!< Contact matrix scaling factor - neighborhood */
+    amrex::Real pWO = amrex::Real(0.5);     /*!< Contact matrix scaling factor - workplace */
+    amrex::Real pHC = amrex::Real(0.5);     /*!< Contact matrix scaling factor - QDG?? */
+    amrex::Real pHCC0 = amrex::Real(0.5);   /*!< Contact matrix scaling factor - QDG?? */
+    amrex::Real pHCC1 = amrex::Real(0.5);   /*!< Contact matrix scaling factor - QDG?? */
```
Comment (debog) [inline src/DiseaseParm.H]: @atmyers @tannguyen153 What are these numbers?

File: src/DiseaseParm.H [hunk 4]
```diff
@@ -1,56 +1,79 @@
...
+    /*! Work transmission probabilities for each age group (<5, 5-17, 18-29, 30-64, 65+) QDG ?? */
     amrex::Real xmit_work = amrex::Real(0.115*pWO);
+    /*! School transmission probabilities for each age group (<5, 5-17, 18-29, 30-64, 65+) QDG ?? */
     amrex::Real xmit_school[7] = {amrex::Real(0.), amrex::Real(.105*pHSC), amrex::Real(.125*pMSC), amrex::Real(.145*pESC), amrex::Real(.145*pESC), amrex::Real(.15), amrex::Real(.35)};
 
     /* Within family, from child or adult, for 5 age groups (here Thai paper, US:0.6->0.8)*/
+    /*! Transmission probability within family for each age-group from adult  QDG ??*/
     amrex::Real xmit_adult[5] = {amrex::Real(.3*pFA), amrex::Real(.3*pFA), amrex::Real(.4*pFA), amrex::Real(.4*pFA), amrex::Real(.4*pFA)};
+    /*! Transmission probability within family for each age-group from adult  QDG ??*/
     amrex::Real xmit_child[5] = {amrex::Real(.6*pFA), amrex::Real(.6*pFA), amrex::Real(.3*pFA), amrex::Real(.3*pFA), amrex::Real(.3*pFA)};
```
Comment (debog) [inline src/DiseaseParm.H:54]: @atmyers @tannguyen153 Are my descriptions correct?

File: src/DiseaseParm.H [hunk 5]
```diff
@@ -1,56 +1,79 @@
...
-    amrex::Real xmit_child_SC[5], xmit_adult_SC[5], xmit_nc_child_SC[5], xmit_nc_adult_SC[5];
-    amrex::Real xmit_comm_SC[5], xmit_hood_SC[5], xmit_sch_c2a[5], xmit_sch_a2c[5];
+    amrex::Real xmit_child_SC[5],     /*!< Transmission probability ... QDG ?? */
+                xmit_adult_SC[5],     /*!< Transmission probability ... QDG ?? */
+                xmit_nc_child_SC[5],  /*!< Transmission probability ... QDG ?? */
+                xmit_nc_adult_SC[5];  /*!< Transmission probability ... QDG ?? */
+    amrex::Real xmit_comm_SC[5],      /*!< Transmission probability ... QDG ?? */
+                xmit_hood_SC[5],      /*!< Transmission probability ... QDG ?? */
+                xmit_sch_c2a[5],      /*!< Transmission probability in school - child to adult QDG ?? */
+                xmit_sch_a2c[5];      /*!< Transmission probability in school - adult to child QDG ?? */
```
Comment (debog) [inline src/DiseaseParm.H:63]: @atmyers @tannguyen153 What are each of these transmission probabilities?

File: src/DiseaseParm.H [hunk 6]
```diff
@@ -1,56 +1,79 @@
...
 
-    amrex::Real infect = amrex::Real(1.0);
+    amrex::Real infect = amrex::Real(1.0); /*!< QDG */
```
Comment (debog) [inline src/DiseaseParm.H:65]: @atmyers @tannguyen153 Is this the default infection probability?

File: src/DiseaseParm.H [hunk 7]
```diff
@@ -1,56 +1,79 @@
...
 
-    amrex::Real reinfect_prob = amrex::Real(0.0);
+    amrex::Real reinfect_prob = amrex::Real(0.0); /*!< QDG */
 
-    amrex::Real vac_eff = amrex::Real(1.0);
+    amrex::Real vac_eff = amrex::Real(1.0); /*!< Vaccine efficacy */
 
-    amrex::Real Child_compliance, Child_HH_closure;
+    amrex::Real Child_compliance, /*!< QDG ?? */
+                Child_HH_closure; /*!< QDG ?? */
```
Comment (debog) [inline src/DiseaseParm.H:72]: @atmyers @tannguyen153 What are these variables?

File: src/IO.cpp [hunk 8]
```diff
@@ -14,8 +18,33 @@ namespace ExaEpi
...
+
+    Writes the current disease spread information and census data (unit, FIPS code, census tract ID,
+    and community number) to a plotfile:
+    + Create an output MultiFab (with the same domain and distribution map as the particle container)
+      with 9 components:
+      + component 0: total
+      + component 1: never infected (#Status::never)
+      + component 2: infected (#Status::infected)
+      + component 3: immune (#Status::immune)
+      + component 4: previously infected (#Status::susceptible QDG??)
```
Comment (debog) [inline src/IO.cpp:31]: @atmyers @tannguyen153 Is the Status::susceptible same as "previously infected" in the plotfile?

File: src/Initialization.cpp [hunk 9]
```diff
@@ -157,6 +196,7 @@ namespace Initialization
                     number = (unsigned int) rint( ((Real) Ndaywork[to]) /
                              ((Real) WG_size * (Start[to+1] - Start[to])) );
 
+                    /* QDG ?? */
                     work_nborhood_ptr[ip]=4*(amrex::Random_int(4, engine))+nborhood_ptr[ip];
```
Comment (debog) [inline src/Initialization.cpp]: @atmyers @tannguyen153 Why is the work neighborhood being set this way?

File: src/Utils.H [hunk 10]
```diff
@@ -11,42 +15,50 @@
...
-    int plot_int;
-    int random_travel_int;
-    short ic_type;
-    std::string census_filename;
-    std::string workerflow_filename;
-    std::string case_filename;
-
-    int aggregated_diag_int;
-    std::string aggregated_diag_prefix;
+    amrex::IntVect size;                /*!< domain size; (QDG: used only for ic_type = demo?)
```
Comment (debog) [inline src/Utils.H:23]: @atmyers @tannguyen153 Is my comment correct (that the domain size specified in the inputs is used only when IC type is demo)?

File: src/CaseData.cpp (hunk 1 above)
Comment (debog) [reply src/CaseData.cpp]: If file has two columns, j is junk, but num_cases2date is not used anyway.

File: src/Initialization.cpp (hunk 9 above)
Comment (atmyers) [reply src/Initialization.cpp]: The idea is that one number stores both 4 unique work neighborhoods and 4 unique home neighborhoods. From this value, you can recover the home neighbor by taking `% 4` and getting the remainder, or you can compute `/ 4` to get a work cluster that is between 0 and 3.

**Comment by atmyers** [comment]
Thanks @debog!

**Pull request merged**